curriculum/challenges/english/07-scientific-computing-with-python/learn-special-methods-by-building-a-vector-space/65f3fa097f26b510db6c710b.md

```diff
@@ -0,0 +1,53 @@
+---
+id: 65f3fa097f26b510db6c710b
+title: Step 9
+challengeType: 20
+dashedName: step-9
+---
+
+# --description--
+
+Great, `norm` is working as expected. Now, if you try to print `v1`, you'll get the default string representation of an object (something like `<__main__.Vector object at 0x11eb778>`).
+
+Inside the `Vector` class, declare an empty `__str__` method to implement a readable string representation. Remember to give it a `self` parameter.
```

Comment:
Absolutely true, but I don't see any easy solution for that. Maybe we can ask to print `v1` in this step to show that if `__str__` does not return a string you'll get a `TypeError`.

If we change the tests so that the code does not run, we can allow this step to pass with the error, but then I don't know how to proceed considering the issue you opened #55037